lib/modules/versioning/ubuntu/index.ts

```diff
@@ -16,13 +16,25 @@ const di = new DistroInfo('data/ubuntu-distro-info.json');
 // validation
 
 function isValid(input: string): boolean {
-  return (
-    (typeof input === 'string' &&
-      regEx(/^(0[4-5]|[6-9]|[1-9][0-9])\.[0-9][0-9](\.[0-9]{1,2})?$/).test(
-        input
-      )) ||
-    di.isCodename(input)
-  );
+  if (typeof input !== 'string') {
+    return false;
+  }
+
+  if (
+    regEx(/^(0[4-5]|[6-9]|[1-9][0-9])\.[0-9][0-9](\.[0-9]{1,2})?$/).test(input)
+  ) {
+    return true;
+  }
+
+  if (di.isCodename(input)) {
+    return true;
+  }
+
+  return isDatedCodeName(input);
+}
+
+function isDatedCodeName(input: string): boolean {
+  return typeof input === 'string' && regEx(/^(\w+)-(\d{8})$/).test(input);
```

Comment:
```suggestion
  return is.nonEmptyString(input) && regEx(/^(\w+)-(\d{8})$/).test(input);
```

Reply:
why do we need this check at all? the function declaration says only string is passed. if it can any other value, then it should accept null and undefined

Reply:
I've left it there as the original function had it there.